src/Stratis.Bitcoin.Features.MemoryPool/MempoolRuleContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool
+{
+    public class MempoolRuleContext
+    {
+        public Network Network { get; }
+
+        public ITxMempool Mempool { get; } 
+
+        public MempoolSettings Settings { get; }
+
+        public ChainIndexer ChainIndexer { get; }
+
+        public IConsensusRuleEngine ConsensusRules { get; }
```

Comment:
I tend to agree that services should not be put in a Context pattern, context it used to create state and allow this state to pass between rules (if needed), rules in gneral should be singletons (they don't keep state).
So what you want to do is inject the services to your rules.
In this case just inject IConsensusRuleEngine,ChainIndexer, MempoolSettings, Network etc... and only in the rules that use them.
It will work well if you create an IMempoolValidator that takes a collection of IMempoolRule (see my PR on injecting rules, maybe use the same pattern?)